src/OpenTelemetry/Sdk.cs

```diff
@@ -41,13 +42,20 @@ static Sdk()
             Activity.DefaultIdFormat = ActivityIdFormat.W3C;
             Activity.ForceDefaultIdFormat = true;
             SelfDiagnostics.EnsureInitialized();
+
+            var sdkVersion = typeof(Sdk).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version;
```

Comment:
Does this include pre-release info? We were including that https://github.com/open-telemetry/opentelemetry-dotnet/pull/4635/files#diff-82711f0e4129a4bdcd2306a09e8ba95ca230d6b911561e8f38461aa242c7b954:~:text=%7D-,private%20static%20string%20GetAssemblyInformationalVersion(),%7D,-%7D

Reply:
Restored! Sorry about that

Reply:
How do we test this? The current test seems to be insufficient.

Reply:
To be fully robust probably needs an integration test that looks at the actual dll's AssemblyInformationalVersionAttribute spit out by the build system? Don't want to tackle that on this PR but I added a unit test that covers the parse logic.